python/ray/dataframe/dataframe.py

```diff
@@ -1186,9 +1186,67 @@ def convert_objects(self, convert_dates=True, convert_numeric=False,
             "github.com/ray-project/ray.")
 
     def corr(self, method='pearson', min_periods=1):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Compute pairwise correlation of columns, excluding NA/null values.
+
+        Arguments:
+            method: {‘pearson’, ‘kendall’, ‘spearman’}
+            min_periods: Minimum number of observations required per pair of
+                columns to have a valid result. Currently only available for
+                pearson and spearman correlation
+
+        Returns:
+            DataFrame of correlations between each pair of columns.
+        """
+
+        if method != 'pearson':
+            raise NotImplementedError(
+                "To contribute to Pandas on Ray, please visit "
+                "github.com/ray-project/ray.")
+
+        # the correlation is only calculated on numerical rows so we will
+        # ignore all other rows
+        new_cols = self.dtypes[self.dtypes.apply(
+            lambda x: is_numeric_dtype(x))].index
+        num_rows = len(self._row_metadata)
+
+        # compute the mean for each numerical column in the DataFrame
+        means = _map_partitions(
+                lambda df: (df.select_dtypes([np.number])
+                            .sum(axis=0)/num_rows).values,
+                self._col_partitions)
+
+        # For each column, subtract the mean for the column from each value in
+        # the column. This will be used in both the covariance and standard
+        # deviation calculations
+        subtracted_means = _map_partitions(
```

Comment:
Is there a way to combine the next two `_map_partitions` calls?

Reply:
we actually need the outputs of both `_map_partitions` calls (we ray.get `subtracted_means` in line 1233 since it's used directly in the correlation computation, in addition to using it as an intermediate value for computing `stdevs`)

Reply:
Can the remote tasks return 2 values instead of using `_map_partitions`? That way it's a single task rather than two tasks.